contracts/Marketplace.sol

```diff
@@ -270,30 +310,45 @@ contract Marketplace is Proofs, StateRetrieval, Endian {
 
     _removeFromMySlots(slot.host, slotId);
 
-    uint256 amount = _requests[requestId].pricePerSlot() +
-      slot.currentCollateral;
-    _marketplaceTotals.sent += amount;
+    uint256 payoutAmount = _requests[requestId].pricePerSlot();
+    uint256 collateralAmount = slot.currentCollateral;
+    _marketplaceTotals.sent += (payoutAmount + collateralAmount);
     slot.state = SlotState.Paid;
-    assert(_token.transfer(slot.host, amount));
+    assert(_token.transfer(slot.payoutAddress, payoutAmount));
+    assert(_token.transfer(slot.host, collateralAmount));
```

Comment:
Not something that has to be done in this PR, but we should consider using `require()` instead of `assert()` as `assert()` [eats all the remaining gas](https://www.evm.codes/#fe?fork=cancun).

Reply:
Good catch! Created an issue to track this: https://github.com/codex-storage/codex-contracts-eth/issues/149